sentry-ruby/spec/sentry_spec.rb

```diff
@@ -784,4 +784,141 @@
     end
   end
 
+  describe ".close" do
+    context "when closing initialized SDK" do
+      it "not initialized?" do
+        expect(described_class.initialized?).to eq(true)
+        described_class.close
+        expect(described_class.initialized?).to eq(false)
+      end
+
+      it "removes main hub" do
+        expect(described_class.get_main_hub).to be_a(Sentry::Hub)
+        described_class.close
+        expect(described_class.get_main_hub).to eq(nil)
+      end
+
+      it "removes thread local" do
+        expect(Thread.current.thread_variable_get(described_class::THREAD_LOCAL)).to be_a(Sentry::Hub)
+        described_class.close
+        expect(Thread.current.thread_variable_get(described_class::THREAD_LOCAL)).to eq(nil)
+
+      end
+
+      it "calls background worker shutdown" do
+        expect(described_class.background_worker).to receive(:shutdown)
+        described_class.close
+        expect(described_class.background_worker).to eq(nil)
+      end
+
+      it "kills session flusher" do
+        expect(described_class.session_flusher).to receive(:kill)
+        described_class.close
+        expect(described_class.session_flusher).to eq(nil)
+      end
+
+      it "disables Tracepoint" do
+        perform_basic_setup do |config|
+          config.capture_exception_frame_locals = true
+        end
+
+        expect(described_class.exception_locals_tp).to receive(:disable)
+        described_class.close
+      end
+
+      it "no-ops top level API getters" do
+        described_class.close
+        expect(described_class.configuration).to eq(nil)
+        expect(described_class.csp_report_uri).to eq(nil)
+        expect(described_class.get_main_hub).to eq(nil)
+        expect(described_class.get_current_hub).to eq(nil)
+        expect(described_class.get_current_client).to eq(nil)
+        expect(described_class.get_current_scope).to eq(nil)
+        expect(described_class.last_event_id).to eq(nil)
+      end
+
+      it "no-ops top level API actions" do
```

Comment:
I think we don't need to test these here. We already have [tests](https://github.com/getsentry/sentry-ruby/blob/master/sentry-ruby/spec/initialization_check_spec.rb) testing uninitialized no-op calls.